projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts

```diff
@@ -152,6 +156,39 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
         this.scrollToBottom();
     }
 
+    private handleQueryParams(): void {
+        this.route.queryParams
+            .pipe(
+                filter((params) => params.id),
+                zipWith(this.userService.getSessionTokenForRequest()),
+                mergeMap(([params, sessionToken]) =>
+                    this.http.get<EbvDataset>(`${this.config.API_URL}/ebv/dataset/${params.id}`, {
+                        headers: new HttpHeaders().set('Authorization', `Bearer ${sessionToken}`),
+                    }),
+                ),
+            )
+            .subscribe((dataset) => {
+                const ebvClass = this.ebvClasses.find((c) => c.name === dataset.ebvClass);
+
+                if (!!ebvClass) {
+                    this.ebvClass = ebvClass;
+                    this.ebvNames = ebvClass.ebvNames;
+
+                    this.ebvName = dataset.ebvName;
+
+                    this.request<Array<EbvDataset>>(`ebv/datasets/${encodeURIComponent(dataset.ebvName)}`, undefined, (data) => {
+                        this.ebvDatasets = data;
+
+                        const selected = data.find((d) => d.id === dataset.id);
+
+                        if (!!selected) {
+                            this.setEbvDataset(selected);
```

Comment:
So, this does not re-assembly https://portal.geobon.org/map/#/?id=1 but stops at the dataset level?

Reply:
efbfc5f59b61d6e75ff375d0188651c370a846cb